apps/confidential/confidential.go

```diff
@@ -283,6 +283,9 @@ type Options struct {
 
 	// Instructs MSAL Go to use an Azure regional token service with sepcified AzureRegion.
 	AzureRegion string
+
+	// Disable instance discovery
+	IsInstanceDiscoveryEnabled bool
```

Comment:
When a field isn't explicitly assigned, its value is its type's default ("zero") value. The zero value of `bool` is `false`. This implementation therefore makes `New(clientID, cred)` the same as `New(clientID, cred, WithInstanceDiscovery(false))` i.e., it disables instance discovery by default.

So, I think `WithInstanceDiscovery` is the right user-facing API, but the internal option must be inverted. Personally, I would do
```go
func WithInstanceDiscovery(enabled bool) Option {
	return func(o *Options) {
		o.DisableInstanceDiscovery = !enabled
	}
}
```

Reply:
I agree with @chlowell 's point on "`WithInstanceDiscovery` is the right user-facing API" and the Instance Discovery behavior shall be enabled by default (to match the historical MSAL behavior).

I'll leave the implementation details to you Go experts to examine `o.DisableInstanceDiscovery = !enabled`. I am having a hard time to understand that implementation, and I would personally do an explicit pseudo code as `if enabled is a true or the default zero; then enable Instance Discovery`.